apps/voting/contracts/Voting.sol

```diff
@@ -361,14 +425,39 @@ contract Voting is IForwarder, AragonApp {
 
     function _canVote(uint256 _voteId, address _voter) internal view returns (bool) {
         Vote storage vote_ = votes[_voteId];
-
         return _isVoteOpen(vote_) && token.balanceOfAt(_voter, vote_.snapshotBlock) > 0;
     }
 
+    function _canVoteOnBehalfOf(uint256 _voteId, address _principal, address _representative) internal view returns (bool) {
+        // TODO: if we support multiple representatives per principal, we will need to store who proxied a vote to allow changing votes
+        return _canVote(_voteId, _principal) &&
+               _isRepresentativeOf(_principal, _representative) &&
+               _hasNotVoteYet(_voteId, _principal) &&
+               !_withinOverruleWindow(_voteId);
+    }
+
+    function _isRepresentativeOf(address _principal, address _representative) internal view returns (bool) {
+        return representatives[_principal][_representative];
+    }
+
+    function _hasNotVoteYet(uint256 _voteId, address _principal) internal view returns (bool) {
+        VoterState state = getVoterState(_voteId, _principal);
+        return state == VoterState.Absent;
+    }
+
     function _isVoteOpen(Vote storage vote_) internal view returns (bool) {
         return getTimestamp64() < vote_.startDate.add(voteTime) && !vote_.executed;
     }
 
+    function _withinOverruleWindow(uint256 _voteId) internal view returns (bool) {
```

Comment:
I would make this one private, as it doesn't have the `_voteExists` check.

Reply:
I fixed it to receive a storage pointer instead